lib/util.lua

```diff
@@ -115,6 +115,19 @@ local function ends_with(str, endstr)
     return str:sub(-#endstr) == endstr
 end
 
+local function dump(t)
+    if type(t) == 'table' then
+       local s = ''
+       for k,v in pairs(t) do
+          if type(k) ~= 'number' then k = '"'..k..'"' end
+          s = s .. '['..k..'] = ' .. dump(v)
+       end
+       return s
+    else
+       return tostring(t) .. '\n'
+    end
+ end
+
```

Comment:
I noticed this new util function, but I find where it is used. Is this a leftover from development or are you planning to use it later?

Reply:
It was for debugging whilr implementing the features, but I figured it's a useful util to leave in for further use.

Reply:
This is redundant since we have [inspect.lua](https://github.com/kikito/inspect.lua) in the repo.

```lua
local inspect = require 'lib/inspect'
print( inspect(my_table) )
```

Reply:
Ah I was not aware. Thank you. I will remove